src/main/webapp/script.js

```diff
@@ -17,6 +17,31 @@ function onSignIn(googleUser) {
     'Welcome ' + profile.getName();
   document.getElementById('profileEmail').innerHTML = profile.getEmail();
   //   document.getElementById('profileId').value = profile.getId();
+
+  signInCallback(id_token);
+}
+
+function signInCallback(authResult) {
+  if (authResult['code']) {
+    // Send the code to the server
+    $.ajax({
+      type: 'POST',
+      url: '/starter',
```

Comment:
I don't understand. Why aren't you posting directly to the `/users` endpoint so you can create the user entity in datastore?

Reply:
I separated the code for the sake of readability because it was a good chunk of code. I then redirected it to the users. Also, the POST is being used to get the information from the Google user (odd how it isn't called a GET, but that's just how they recommended it) first. Then I add those parameters to the request and use that as a post to create the user in the datastore.

Reply:
It's probably called a POST because you're expected to do something with the token and create something in the backend. Since you already have a `POST /users` endpoint and already have the user info here though, you can make the `POST /users` API call here directly instead of rerouting the request in backend using a new endpoint.